### PR TITLE
Adjust expressions and driver supported features

### LINKED ISSUE #12598: SQLite does not support Right or Full Outer joins
**Describe the bug**
SQLite does not support Right or Full Outer joins, but it's available options in Notebook editor, and [documentation](https://www.metabase.com/docs/latest/users-guide/custom-questions.html#joining-data) incorrectly says it does.

**To Reproduce**
1. Custom question > select SQLite database and table
2. Join any table
3. Click "Change join type" (icon between the two tables) and select "Right outer join" or "Full outer join"
image
4. Query will fail with `SQL error or missing database (RIGHT and FULL OUTER JOINs are not currently supported)`

**Expected behavior**
Should not be able to select those joins

**Information about your Metabase Installation:**
Metabase 0.35.3 and `master`

**Additional context**
SQLite limitation: https://www.sqlite.org/omitted.html
Workaround for databases not supporting https://en.wikipedia.org/wiki/Join_(SQL)#Full_outer_join

### PR DESCRIPTION
**Pull request by flamber**
This started out as a Fixes #12598, but then noticed that several of the expression docs limitations were not equal to the drivers.

1. BigQuery seems to have limited `:percentile-aggregations` in [0.35.4](https://github.com/metabase/metabase/blame/27c79d2beeccee5541fe83b87cd5b07b8d9f8742/modules/drivers/bigquery/src/metabase/driver/bigquery.clj#L251), but I can still write `Median([column])` (similar to #12222), which "works", but generates incorrect SQL missing ticks around `dataset.table`, so the query fails.
I'm not sure if the limitation was intended in 0.35.4. Also not sure why `:expressions` is limited.
2. Not sure why Presto seems to [list supported features](https://github.com/metabase/metabase/blob/1375b4991c4bd18673e04831ecae6415854c6487/modules/drivers/presto/src/metabase/driver/presto.clj#L404-L412) with `true`, where the other driver only list limitations with `false`. ~~Haven't tested, but I guess it supports `:percentile-aggregations`, when it's not defined with `false` explicitly.~~ Have tested, and only `Median` works - `Percentile` fails.
3. ~~Haven't tested Vertica, just removed the limitation.~~ Created #12845 and add the limitation back

### CODE REVIEW

**Review by jeff-bruemmer — CHANGES_REQUESTED** [review]
The **Right outer join** description is inaccurate. Should read: "select all records from Table B, along with records from Table A that meet the join condition, if any."

**Change:** "By default, Metabase will do a left outer join, but you can click on the Venn diagram icon to change this to a different type of join. Not all databases supports all types of joins, so the options you'll see will differ based on the type of database you're using. Here are what the basic types of joins each do:"

**To**: "By default, Metabase will do a left outer join, but you can click on the Venn diagram icon to select a different type of join. Not all databases support all types of joins, so Metabase will only display the options supported by the database you're using. 

Here are the basic types of joins:"

Optional: if join operation support is a common question: consider organizing join functionality in a table to clearly indicate which databases support which operations, with join operations as columns, and databases as rows.

**Comment by sbelak** [comment]
When touching drivers code, it's good to run driver tests by appending `[ci drivers]` to the commit message.

(on a related note, I think Vertica has percentile aggregations disabled, because I couldn’t get them to work for some reason. I'll look into it again)

**Review by jeff-bruemmer — APPROVED** [review]

File: modules/drivers/vertica/src/metabase/driver/vertica.clj [hunk 1]
```diff
@@ -22,9 +22,6 @@
 
 (driver/register! :vertica, :parent #{:sql-jdbc ::legacy/use-legacy-classes-for-read-and-set})
 
-(defmethod driver/supports? [:vertica :percentile-aggregations] [_ _] false)
```
Comment (camsaul) [inline modules/drivers/vertica/src/metabase/driver/vertica.clj:25]: Does Vertica support percentile aggregations? @sbelak would we be breaking anything by removing this?

Comment (sbelak) [reply modules/drivers/vertica/src/metabase/driver/vertica.clj:25]: Supposedly yes, but I couldn’t get it to working following the example in their documentation. I think it's best to leave it off for now, and I'll open a new issue for it.

**Comment by flamber** [comment]
I have reverted the Vertica limitation (and opened #12845). Now it's down to 2 questions:
1. BigQuery - should it have the following as **not** supported, since `:expressions` is limited: `abs`, `floor`, `ceil`, `round`
2. Presto - should it have `:percentile-aggregations` limited, since `median` works, but `percentile` fails (in my small test)

**Comment by sbelak** [comment]
> 2. Presto - should it have `:percentile-aggregations` limited, since `median` works, but `percentile` fails (in my small test)
That's odd, as they use the same function underneath. I'll have a look

**Comment by sbelak** [comment]
> 2. Presto - should it have `:percentile-aggregations` limited, since `median` works, but `percentile` fails (in my small test)
That's odd, as they use the same function underneath. I'll have a look

File: docs/users-guide/expressions.md [hunk 2]
```diff
@@ -94,21 +94,18 @@ This would return rows where `Created At` is between January 1, 2020 and March 3
...
-- sqrt
-- variance
+**SQL Server**: `median`, `percentile` and `regexextract`
 
-Additionally, **Vertica**, **BigQuery**, and **Presto** only provide _approximate_ results for median and percentile.
+**SQLite**: `log`, `median`, `percentile`, `power`, `regexextract`, `standardDeviation`, `sqrt` and `variance`
+
+**Vertica**: `median` and `percentile`
+
+Additionally, **Presto** only provide _approximate_ results for `median` and `percentile`.
```
Comment (mazameli) [inline docs/users-guide/expressions.md:109]:
```suggestion
Additionally, **Presto** only provides _approximate_ results for `median` and `percentile`.
```

**Comment by mazameli** [comment]
@flamber, this seems g2g besides that typo I just noted and fixed. Shall we merge this once tests finish?

**Comment by flamber** [comment]
@mazameli Let me block the expression functions on BigQuery, since the driver says they're limited (probably a hack to have Custom Column not show for that database, from what I can understand)

File: docs/users-guide/expressions.md [hunk 3]
```diff
@@ -94,21 +94,18 @@ This would return rows where `Created At` is between January 1, 2020 and March 3
 
 Certain database types don't support some of the above functions:
 
-MySQL and SQL Server
+**BigQuery**: `median` and `percentile`
```
Comment (flamber) [inline docs/users-guide/expressions.md:97]:
```suggestion
**BigQuery**: `abs`, `ceil`, `floor`, `median`, `percentile` and `round`
```

**Pull request merged**